clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -38,11 +39,15 @@
  */
 public class MemoryRecordsBuilder {
     private static final float COMPRESSION_RATE_ESTIMATION_FACTOR = 1.05f;
+    private static final DataOutputStream CLOSED_STREAM = new DataOutputStream(new OutputStream() {
+        @Override
+        public void write(int b) throws IOException {
+            throw new IllegalStateException("MemoryRecordsBuilder is closed for record appends");
+        }
+    });
```

Comment:
Out of curiosity, why use this instead of just setting it to `null`?

Reply:
Since I removed the flag which was tracking whether the stream was closed, I thought it will be better to throw a meaningful exception rather than NPE if we made any changes later which missed out the check for `appendStreamIsClosed`.